prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -1,17 +1,20 @@
 import os
 
 from dotenv import load_dotenv
-from functions import ALL_FUNCTIONS
+from functions import MARKET_FUNCTIONS, MISC_FUNCTIONS
 from microchain import LLM, Agent, Engine, OpenAIChatGenerator
 from microchain.functions import Reasoning, Stop
+from prediction_market_agent_tooling.markets.markets import MarketType
 
 load_dotenv()
 
 engine = Engine()
 engine.register(Reasoning())
 engine.register(Stop())
-for function in ALL_FUNCTIONS:
+for function in MISC_FUNCTIONS:
     engine.register(function())
+for function in MARKET_FUNCTIONS:
+    engine.register(function(market_type=MarketType.OMEN))
```

Comment:
I was wondering if we accomplish this via a decorator, e.g. we mark all classes that are market functions with @MarketFunction and retrieve those afterwards.
I created a quick-and-dirty example here -> https://gist.github.com/gabrielfior/3d777af14080a121a0cefa2ee2f172e9
The current implementation is fine, but we might forget to keep those lists updated.

Reply:
> https://gist.github.com/gabrielfior/3d777af14080a121a0cefa2ee2f172e9

That's an interesting trick!

No preference from my side but it's cool

Reply:
Oo yes nice trick. I think we need to have market-specific functions inherit from `MarketFunction` anyway though. Feels a bit clunky to me registering them separately like this, or can this be combined?

```
@market_function
class GetBalance(MarketFunction)
    ...
```

I know you can get subclasses of a class with:
```
import inspect

def find_imported_classes_by_type(base_class):
    classes = []
    for name, obj in globals().items():
        if inspect.isclass(obj) and issubclass(obj, base_class) and obj is not base_class:
            classes.append(obj)
    return classes

MARKET_FUNCTIONS = find_imported_classes_by_type(MarketFunction)
```

but then you have the issue where 'abstract' subclasses (i.e. `BuyTokens` in this case) will be added to `MARKET_FUNCTIONS`.